web/src/main/java/se/inera/intyg/webcert/web/service/facade/modal/confirmation/ConfirmationModalProviderResolver.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2024 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.service.facade.modal.confirmation;
+
+import se.inera.intyg.common.db.support.DbModuleEntryPoint;
+import se.inera.intyg.common.support.facade.model.CertificateStatus;
+
+public class ConfirmationModalProviderResolver {
+
+    private ConfirmationModalProviderResolver() {
+
+    }
+
+    public static ConfirmationModalProvider get(String type, CertificateStatus status, String origin, boolean isCreatedFromList) {
+        final var isIntegratedOrigin = origin.equals("DJUPINTEGRATION");
+
+        if (!isIntegratedOrigin && !isCreatedFromList) {
```

Comment:
The usage of neglects is confusing, but I think I understand the purpose of this if-statement. To my understanding its so that we can handle both integrated and "not integrated" user cases. Does it make sense to instead separate the normal/integrated by different methods?

If its not possible to separate, maybe a describing method name would clarify the purpose of the if-statement?

Do we also need to consider revision? Otherwise this modal will be displayed for the user every time they open a DB, maybe its like that already? Just checking.

Reply:
Integrated -> The modal is open everytime the draft is opened
Not integrated -> The modal is opened when the draft is created from the list (Modalen visas endast en gång. Har en användare bockat i att kontroll är gjord visas inte modalen igen i fristående Webcert)

Revision is not relevant since we don't have a requirement saying do apply some requirement before the draft has been edited. 

I can rename the neglect but I don't think it's confusing. It states if the origin is not integrated and the draft is not created from list then apply rule.

Reply:
https://inera.atlassian.net/wiki/spaces/IT/pages/9075408/D3.1.2.+Skatteverkets+och+Socialstyrelsen+intyg